src/support.cpp

```diff
@@ -48,6 +48,261 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+    //     It means that we can have at most 5 different densities, each is 1/2 than the other. So, it will looks like below:
+    //           |  step  |    density    |  description              |
+    //           |    0   |   1 / (2^0)   |  100% the most dense      |
+    //           |    1   |   1 / (2^1)   |                           |
+    //           |    2   |   1 / (2^2)   |                           |
+    //           |    3   |   1 / (2^3)   |                           |
+    //           |    4   |   1 / (2^4)   |                           |
+    //           |    5   |   1 / (2^5)   |  3.125% the least dense   |
+    //
+    //  -> With those layer groups, we can project areas with different densities onto the layer we are currently processing.
+    //     Like in the illustration above, you can see 4 different areas with densities ranging from 0 to 3.
+    //
+    //  -> We save those areas with different densities into the SupportInfillPart data structure, which holds all the support infill
+    //     information of a support island on a specific layer.
+    //
+    //  -> Note that this function only does the above, which is identifying and storing support infill areas with densities.
+    //     The actual printing part is done in FffGcodeWriter.
+    //
+
+    // no early-out for this function; it needs to initialize the [infill_area_per_combine_per_density]
+    float layer_skip_count = 8; // skip every so many layers as to ignore small gaps in the model making computation more easy
+    unsigned int gradual_support_step_layer_count = round_divide(gradual_support_step_height, storage.getSettingInMicrons("layer_height")); // The difference in layer count between consecutive density infill areas
+
+    // make gradual_support_step_height divisable by layer_skip_count
+    float n_skip_steps_per_gradual_step = std::max(1.0f, std::ceil(gradual_support_step_layer_count / layer_skip_count)); // only decrease layer_skip_count to make it a divisor of gradual_support_step_layer_count
+    layer_skip_count = gradual_support_step_layer_count / n_skip_steps_per_gradual_step;
+
+    size_t min_layer = 0;
+    size_t max_layer = total_layer_count - 1;
+
+    const ExtruderTrain& infill_extr = *storage.meshgroup->getExtruderTrain(storage.getSettingAsIndex("support_infill_extruder_nr"));
+    const coord_t support_line_width = infill_extr.getSettingInMicrons("support_line_width");
+    const int wall_line_count = 1;  // the wall line count is used for calculating insets, and we generate support infill patterns within the insets
+
+    for (unsigned int layer_nr = 0; layer_nr < total_layer_count - 1; ++layer_nr)
+    {
+        assert(storage.support.supportLayers[layer_nr].support_infill_parts.empty() && "support infill part list is supposed to be uninitialized");
+
+        // this is the complete support areas on this layer
+        const Polygons& whole_support_areas = storage.support.supportLayers[layer_nr].supportAreas;
```

Comment:
The support areas are now contained twice in the `SliceDataStorage`: once in `SupportLayer::supportAreas` and once in `SupportInfillPart::outline`.

I think you should make it so that the `supportAreas` are already `splitIntoParts` in the normal `AreaSupport::generateSupportAreas` and remove `SupportLayer::supportAreas` and instead save the result immediately in `SupportInfillPart::outline`.